Lib/test/test_posix.py

```diff
@@ -1689,6 +1693,37 @@ def test_link_dir_fd(self):
             self.assertEqual(posix.stat(fullname)[1],
                 posix.stat(fulllinkname)[1])
 
+    @unittest.skipIf(
+        support.is_wasi,
+        "WASI: symlink following on path_link is not supported"
+    )
+    @unittest.skipUnless(
+        hasattr(os, "link") and os.link in os.supports_fd,
+        "test needs fd support in os.link()"
+    )
+    @unittest.skipUnless(root_in_posix,
+        "requires the CAP_DAC_READ_SEARCH capability")
```

Comment:
On my Fedora 42, I can use AT_EMPTY_PATH as a regular user. I don't need to be root.

Reply:
It does not look like your shell has such capability. Maybe Fedora has patches that ignores it?